packages/grid/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationFunctions.ts

```diff
@@ -2,12 +2,12 @@ import { GridValueFormatterParams } from '@mui/x-data-grid-pro';
 import { isNumber } from '@mui/x-data-grid-pro/internals';
 import { GridAggregationFunction } from './gridAggregationInterfaces';
 
-const sumAgg: GridAggregationFunction<number> = {
+const sumAgg: GridAggregationFunction<unknown, number> = {
   apply: ({ values }) => {
     let sum = 0;
     for (let i = 0; i < values.length; i += 1) {
       const value = values[i];
-      if (value != null) {
+      if (isNumber(value)) {
```

Comment:
We should only consider numeric values here, otherwise, string concatenation kicks in 😅